### DISCUSSION TITLE
Is there a way to tell gitleaks that a given line is not a leak?

### DISCUSSION THREAD

**Discussion by anzelpwj** [zricethezav/gitleaks#603] [Q&A]
Let's say I have a non-secret in my code, e.g.

```python
DUMMY_PASSWORD = "9f4c8968bee6e5a22b61a3dcd27ab704"
test_password(DUMMY_PASSWORD)
```
I've just made up this key, so I know it's not a leaked password. If I was using something like bandit, I could use the `# nosec` comment on that line to tell Bandit this is not a flaw. Does GitLeaks have a similar command I can use?

**Comment by w0rmr1d3r** [comment]
Hello @anzelpwj !
Have you taken a look at this -> https://github.com/zricethezav/gitleaks#rules-summary ?

**Reply by anzelpwj** [reply]
I've seen the rule list, but I'm curious if there's anything inline that I can do - especially if I want to have a generic ruleset for multiple repos.

**Reply by w0rmr1d3r** [reply]
Hello @anzelpwj , what do you mean by `inline`?
For the multiple repos, they can share the same configuration file

**Reply by anzelpwj** [reply]
If I was using the `bandit` scanner for Python, I might have some code like
```python
sql_query = f"SELECT * FROM {schema_name}.table"
conn.execute(sql_query)
```
And bandit would flag the `sql_query` line as a potential SQL injection vulnerability. So I go and clear up the vulnerability (e.g. `schema_name` is hard-coded to have only a few values) and I need to tell bandit that this is no longer a real vulnerability. I would do this by adding a `# nosec` comment to the code.
```python
sql_query = f"SELECT * FROM {schema_name}.table"  # nosec
conn.execute(sql_query)
```
This is an inline comment to tell bandit that this particular vulnerability is no longer a concern.

**Reply by w0rmr1d3r** [reply]
Oh I see!
For now, detecting vulnerabilities is not a feature.
By adding comments to ignore secrets on particular lines is not possible at the moment, that's why I pointed out to use a custom configuration file.

**Reply by anzelpwj** [reply]
Sure, wasn't planning on using this for detecting vulnerabilities, just wanted to have a good way to deal with false positives. I'd really like to use the min (and max?) entropy ruleset, but I'm worried that if I don't get a good minimum entropy setting, I might get a lot of rejections.

**Reply by stephenmbarnes** [reply]
I have the same question/concern.   I'd like to have a very simple way to ignore false positives and/or token/placeholder values in scenarios where a secret only gets inserted in the file at deployment time (yes, I'm aware of the risks of this practice, but it's still an improvement).

**Reply by zricethezav** [reply]
@stephenmbarnes @anzelpwj how about # glignore

This can be accomplished now if you add `#glignore` to the global allowlist regex. I'll look into adding this as a default

**Reply by anzelpwj** [reply]
Apologies for not responding this earlier, but `# glignore` would be exactly what I'm looking for!

**Reply by maltemorgenstern** [reply]
Hey @anzelpwj @stephenmbarnes
I think this feature now exists: https://github.com/zricethezav/gitleaks#gitleaksallow

Or - if you want to ignore specific findings without adding comments - you might want to take a look at https://github.com/zricethezav/gitleaks#gitleaksignore
